lib/spack/spack/modules/common.py

```diff
@@ -930,13 +930,22 @@ def remove(self):
         if os.path.exists(mod_file):
             try:
                 os.remove(mod_file)  # Remove the module file
+                self.remove_module_defaults()  # Remove default targeting module file
                 os.removedirs(
                     os.path.dirname(mod_file)
                 )  # Remove all the empty directories from the leaf up
             except OSError:
                 # removedirs throws OSError on first non-empty directory found
                 pass
 
+    def remove_module_defaults(self):
+        if any(self.spec.satisfies(default) for default in self.conf.defaults):
+            # This spec matches a default, symlink needs to be removed as we
+            # remove the module file it targets.
+            default_symlink = os.path.join(os.path.dirname(self.layout.filename), "default")
+            if os.path.lexists(default_symlink):
+                os.remove(default_symlink)
+
```

Comment:
```suggestion
    def remove_module_defaults(self):
        if not any(self.spec.satisfies(default) for default in self.conf.defaults):
            return

        # This spec matches a default, symlink needs to be removed as we remove the module
        # file it targets.
        default_symlink = os.path.join(os.path.dirname(self.layout.filename), "default")
        try:
            os.unlink(default_symlink)
        except OSError:
            pass

```

I would just attempt to delete the file (and ignore errors)